File: Two Sum/ans.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> seen;
        for (int i =0; i < nums.size();i++){
            int diff = target - nums[i];
            if (seen.find(diff) != seen.end()){
                return{seen[diff], i};
            }else{
                seen[nums[i]] = i;
            }

        }
        return {-1,-1};

        
    }
};

File: Single Number/ans.py
''''
Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.

You must implement a solution with a linear runtime complexity and use only constant extra space.
'''
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0 
        for num in nums:
            ans = ans ^ num
        return ans
        
